src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -740,6 +740,12 @@ private void AddSegment(int sizeHint = 0)
             _currentSegment = owner.Memory;
             _currentSegmentOwner = owner;
         }
+        else if (sizeHint <= MemoryPool<byte>.Shared.MaxBufferSize)
```

Comment:
nit: Reduce duplication by adding a static method and call it twice. e.g.

```
static bool TryRentBuffer(MemoryPool<byte> pool, int sizeHint, ref Memory<byte> buffer, ref IMemoryOwner<byte>? owner)
{
    if (sizeHint <= pool.MaxBufferSize)
    {
        owner = pool.Rent(sizeHint);
        buffer = owner.Memory;
        return true;
    }

    return false;
}
```

Then call with `_memoryPool` and `MemoryPool<byte>.Shared`.